pkg/controllers/monitoring-stack/controller.go

```diff
@@ -173,7 +172,7 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 func (r *reconciler) deleteGrafanaDS(ctx context.Context, ms *stack.MonitoringStack) error {
 	logger := r.logger.WithValues(stackName(ms)...)
 
-	gds := types.NamespacedName{Namespace: goctrl.Namespace, Name: GrafanaDSName(ms)}
+	gds := types.NamespacedName{Namespace: ms.Namespace, Name: GrafanaDSName(ms)}
```

Comment:
I think the operator shouldn't create grafana datasources. Also not the grafana data source need to be created in the same namespace as the grafana operator, so it best to let users using Grafana Operartor deal with that.